mlflow/metrics/utils/make_genai_metric.py

```diff
@@ -37,6 +38,39 @@ def _format_variable_string(variables: Dict[str, Any], eval_df, indx) -> str:
     )
 
 
+# Function to extract Score and Justification
+def _extract_score_and_justification(output):
+    if (
+        isinstance(output, dict)
+        and "candidates" in output
+        and isinstance(output["candidates"], list)
+        and output["candidates"]
+    ):
+        text = output["candidates"][0]["text"]
+    else:
+        text = output.candidates[0].text
+
+    if text:
+        # Attempt to parse JSON
+        try:
+            data = json.loads(text)
+            score = data.get("Score")
+            justification = data.get("Justification")
```

Comment:
i think we should keep the checks for `isinstance` on score and justification. it is possible that the response is a valid json but not the right type.